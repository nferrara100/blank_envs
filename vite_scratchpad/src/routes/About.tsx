import * as React from "react";

export function About() {
  return (
    <div>
      <h2>About</h2>
      <p>Learn more about us.</p>
    </div>
  );
}
