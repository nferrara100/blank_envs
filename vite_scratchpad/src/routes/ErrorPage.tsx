import * as React from "react";
import { Link } from "react-router-dom";

export default function ErrorPage() {
  return (
    <div>
      <h2>Oops!</h2>
      <p>
        Page not found. Please go <Link to="/">home</Link> and try again.
      </p>
    </div>
  );
}
