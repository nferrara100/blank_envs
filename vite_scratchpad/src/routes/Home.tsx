import * as React from "react";

export function Home() {
  return (
    <div>
      <h2>Home</h2>
      <p>Welcome home!</p>
    </div>
  );
}
