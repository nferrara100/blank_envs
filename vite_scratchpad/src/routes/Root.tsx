import * as React from "react";
import { Link, Outlet } from "react-router-dom";

export function Root() {
  return (
    <main className="container">
      <nav>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/about">About</Link>
          </li>
        </ul>
      </nav>

      <hr />

      <Outlet />
    </main>
  );
}
