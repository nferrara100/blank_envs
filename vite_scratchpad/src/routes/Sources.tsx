import { useMutation } from "@tanstack/react-query";
import { useSources } from "../queries/useSources";

export function Sources() {
  const { isPending, error, data } = useSources();
  // const [text, setText] = useState();

  const mutation = useMutation({
    mutationFn: (data) => {
      return fetch("http://localhost:3000/comments/", {
        method: "post",
        body: JSON.stringify(data),
        headers: { "Content-Type": "application/json" },
      }).then((res) => res.json());
    },
  });

  if (isPending)
    return <button aria-busy="true" className="secondary"></button>;

  if (error) return <article>An error has occurred {error.message}</article>;

  const translations = [];
  for (const translation of data.translations) {
    translations.push(<li>{translation.text}</li>);
  }

  const comments = [];
  for (const comment of data.comments) {
    translations.push(<li>{comment.text}</li>);
  }

  const submitComment = async (e) => {
    e.preventDefault();
    const formData = new FormData(e.target);
    mutation.mutate({ sourceId: 1, text: formData.get("text") });
  };

  return (
    <div>
      <h1>{data.text}</h1>
      <ul>{translations}</ul>
      <ul>{comments}</ul>

      {mutation.isPending ? (
        <p>Submitting</p>
      ) : (
        <>
          {mutation.isError ? (
            <div>An error occurred: {mutation.error.message}</div>
          ) : null}

          {mutation.isSuccess ? <div>Comment added!</div> : null}
        </>
      )}

      <form onSubmit={submitComment}>
        <input type="text" name="text"></input>
        <button type="submit">Submit</button>
      </form>
    </div>
  );
}
