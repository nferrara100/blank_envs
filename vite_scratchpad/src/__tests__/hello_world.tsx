import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Home } from "../routes/Home";

function sum(a, b) {
  return a + b;
}

test("adds 1 + 2 to equal 3", () => {
  expect(sum(1, 2)).toBe(3);
});

test("home displays", () => {
  const queryClient = new QueryClient();

  const component = renderer.create(
    <QueryClientProvider client={queryClient}>
      <Home />
    </QueryClientProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
