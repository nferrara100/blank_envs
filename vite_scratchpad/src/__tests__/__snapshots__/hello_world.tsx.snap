// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home displays 1`] = `
<button
  aria-busy="true"
  className="secondary"
/>
`;
